lib/stimulus_reflex/broadcasters/selector_broadcaster.rb

```diff
@@ -6,9 +6,14 @@ def broadcast(_, data = {})
       morphs.each do |morph|
         selectors, html = morph
         updates = selectors.is_a?(Hash) ? selectors : Hash[selectors, html]
-        updates.each do |selector, html|
+        updates.each do |key, value|
+          updates[key] = reflex.render(key) if key.is_a?(ActiveRecord::Base) && value.nil?
+          updates[key] = reflex.wrap(reflex.render(key), key) if key.is_a?(ActiveRecord::Relation) && value.nil?
+        end.transform_keys { |key|
+          key.is_a?(ActiveRecord::Base) || key.is_a?(ActiveRecord::Relation) ? reflex.dom_id(key) : key
+        }.each do |selector, html|
```

Comment:
We could eliminate a loop here by moving the `transform_keys` logic into the `each`.

```ruby
.each do |key, html|
  selector = key.is_a?(ActiveRecord::Base) || key.is_a?(ActiveRecord::Relation) ? reflex.dom_id(key) : key
  
  ...
```

Reply:
Actually, I'm not sure that you can!

The whole reason I did this is that Ruby complains about adding keys during enumeration.

If you do it with `each`, you're essentially creating a new key and deleting the old one. `transform_keys` actually does what it suggests.

If you know of a way around this, I only learned all of this two days ago. I'm confident you could have a better answer. I just wanted you to know that this was an intentional thing.

Reply:
I haven't tested this yet, but I suspect that we can refactor broadcast into this.

```ruby
def broadcast(_, data = {})
  morphs.each do |morph|
    selectors, html = morph
    updates = selectors.is_a?(Hash) ? selectors : Hash[selectors, html]

    updates.each do |key, value|
      html = reflex.render(key) if key.is_a?(ActiveRecord::Base) && value.nil?
      html = reflex.wrap(reflex.render(key), key) if key.is_a?(ActiveRecord::Relation) && value.nil?
      fragment = Nokogiri::HTML.fragment(html.to_s)

      selector = key.is_a?(ActiveRecord::Base) || key.is_a?(ActiveRecord::Relation) ? reflex.dom_id(key) : key
      match = fragment.at_css(selector)

      if match.present?
        operations << [selector, :morph]
        cable_ready.morph(
          selector: selector,
          html: match.inner_html,
          children_only: true,
          permanent_attribute_name: permanent_attribute_name,
          stimulus_reflex: data.merge({
            morph: to_sym
          })
        )
      else
        operations << [selector, :inner_html]
        cable_ready.inner_html(
          selector: selector,
          html: fragment.to_html,
          stimulus_reflex: data.merge({
            morph: to_sym
          })
        )
      end
    end
  end

  cable_ready.broadcast
  morphs.clear
end
```

Reply:
There were a few small tweaks (eg what gets passed to the fragment has to fall back to an empty string, now) but this worked. I'm always thrilled when you see these optimizations.